### PR TITLE
CLN: use .view(i8) instead of .astype(i8) for datetimelike values

### PR DESCRIPTION
**Pull request by jbrockmendel**
prelude to deprecating the astype behavior

### CODE REVIEW

File: pandas/core/window/ewm.py [hunk 1]
```diff
@@ -252,7 +253,9 @@ def __init__(
                 raise ValueError(
                     "halflife must be a string or datetime.timedelta object"
                 )
-            self.times = np.asarray(times.astype(np.int64))
+            if isna(times).any():
+                raise ValueError("Cannot convert NaT values to integer")
```
Comment (mroeschke) [inline pandas/core/window/ewm.py:257]: Could you add a test for this case?

**Comment by jreback** [comment]
any perf benefits?

**Comment by jbrockmendel** [comment]
> any perf benefits?

maybe some call overhead, but the existing astype call is already doing view under the hood (which is not great)

File: pandas/core/window/ewm.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/window/ewm.py:257]: double-checked and you're right we dont have any.  suggestions for what this might look like?

Comment (mroeschke) [reply pandas/core/window/ewm.py:257]: Something simple like

```
with pytest.raises(ValueError, match=...):
     Series(range(1)).ewm(com=0.1, times=to_datetime(['NaT'])))
```

works.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**